src/app/fdctl/run/tiles/metric.seccomppolicy

```diff
@@ -1,32 +1,69 @@
+
+# logging: all log messages are written to a file and/or pipe
+#
+# 'WARNING' and above are written to the STDERR pipe, while all messages
+# are always written to the log file.
+#
+# arg 0 is the file descriptor to write to.  The boot process ensures
+# that descriptor 2 is always STDERR and descriptor 4 is the logfile.
+write: (or (eq (arg 0) 2)
+           (eq (arg 0) logfile_fd))
 
 # logging: 'WARNING' and above fsync the logfile to disk immediately
 #
 # arg 0 is the file descriptor to fsync.
 fsync: (eq (arg 0) logfile_fd)
 
-# server: serving metric values over HTTP requires accepting connections
+# server: serving pages over HTTP requires accepting connections
 #
-# arg 0 is the listen socket file descriptor to accept connections on
-accept: (and (eq (arg 0) socket_fd)
-             (eq (arg 1) 0)
-             (eq (arg 2) 0))
+# arg 0 is the listen socket file descriptor to accept connections on.
+accept4: (and (eq (arg 0) metrics_socket_fd)
+              (eq (arg 1) 0)
+              (eq (arg 2) 0)
+              (eq (arg 3) "SOCK_CLOEXEC|SOCK_NONBLOCK"))
 
-# server: serving metric values over HTTP requires reading from conns
-read
+# server: serving pages over HTTP requires reading from connections
+#
+# arg 0 is the file descriptor to read from.  It can be any of the
+# connected client sockets returned by accept4(2).  To accomodate this,
+# we allow any file descriptor except those which we know are not these
+# connected clients, which are the log file, STDOUT, and the listening
+# socket itself.
+read: (not (or (eq (arg 0) 2)
+               (eq (arg 0) logfile_fd)
+               (eq (arg 0) metrics_socket_fd)))
 
-# server: serving metric values over HTTP requires writing to conns
-write
+# server: serving pages over HTTP requires writing to connections
+#
+# arg 0 is the file descriptor to send to.  It can be any of the
+# connected client sockets returned by accept4(2).  To accomodate this,
+# we allow any file descriptor except those which we know are not these
+# connected clients, which are the log file, STDOUT, and the listening
+# socket itself.
+sendto: (not (or (eq (arg 0) 2)
+                 (eq (arg 0) logfile_fd)
+                 (eq (arg 0) metrics_socket_fd)))
 
-# server: serving metric values over HTTP requires closing conns
-close
+# server: serving pages over HTTP requires closing connections
+#
+# arg 0 is the file descriptor to close.  It can be any of the connected
+# client sockets returned by accept4(2).  To accomodate this, we allow
+# any file descriptor except those which we know are not these connected
+# clients, which are the log file, STDOUT, and the listening socket
+# itself.
+close: (not (or (eq (arg 0) 2)
+                (eq (arg 0) logfile_fd)
+                (eq (arg 0) metrics_socket_fd)))
 
-# server: serving metric values over HTTP requires polling conns
-poll
+# server: serving pages over HTTP requires polling connections
+#
+# arg 2 is the timeout.
+poll: (eq (arg 2) 0)
```

Comment:
Do you want to do a similar filter on the file descriptor as for `sendto`, etc. here?

Reply:
I don't think we can with the current filter implementation, since you provide a list of `pollfd` structs. See `man 2 poll`

Reply:
Oh good point. I forgot what the syscall looked like.